itwm_example/mco/mco.py

```diff
@@ -118,109 +77,114 @@ def optimize(self):
         return optimal_point, optimal_kpis
 
 
-def opt(weighted_score_func, initial_point, constraints):
-    """Partial func. Performs a scipy optimise with SLSQP method given the
-    scoring function, the initial point, and a set of constraints."""
-
-    return optimize.minimize(
-        weighted_score_func,
-        initial_point,
-        method="SLSQP",
-        bounds=constraints).x
-
-
-def get_weight_combinations(dimension, num_points, zero_values=True):
-    """Given the number of dimensions, this function provides all possible
-    combinations of weights adding to 1.0. For example, a dimension 3
-    will give all combinations (x, y, z) where x+y+z = 1.0.
-
-    The num_points parameter indicates how many divisions along a single
-    dimension will be performed. For example num_points == 3 will evaluate
-    for x being 0.0, 0.5 and 1.0. The returned (x, y, z) combinations will
-    of course be much higher than 3.
+class MCO(BaseMCO):
 
-    Note that if the zero_values parameter is set to false, then ensure
-    num_points > dimension in order for the generator to return any values.
+    optimizer = Type(IOptimizer)
```

Comment:
A good question to ask at this point would be could we convert this into an `Instance` trait? Review what we are gaining by using a `Type` trait here, and whether it is more appropriate than having `self.optimizer` as an instance.

Reply:
I don't really think that having `optimizer` as an actual instance won't have some "bad" effects in a long run. I suggest keeping it as it is, and fix the existing multiple initialization of `optimizer`s, where only one instance is required.